cluster/variables.tf

```diff
@@ -123,6 +123,8 @@ variable "labels" {
 
 variable "security_group" {
```

Comment:
To be honest, this resource accepts one sec group email address. However to use the dynamic blocks, I could not see an option for a string parameter. Therefore it is list now. In the cloud infrastructure setup repo, it will also be a list but I'll use just one string variable to assign the value for that. Not ideal, but still solves the issue :)

Reply:
Actually this seems better and now I tried it but ended up with issues. When I use [var.security_group], it still looks for security_group variable and even if I set it to "" or null, it still activates the dynamic block which leads to the original issue which I want to resolve.
I only want this block to be activated when I set security_group. With the list which has default [], it acts like there is nothing if I do not set anything (which is fine), but with the string type and [var.security_group] it is not working.

Reply:
I'm merging like this since you both approved and since this is the only solution I have, feel free to ping me if there is more elegant way to achieve this